src/Microsoft.AspNetCore.SignalR.Redis/RedisHubLifetimeManager.cs

```diff
@@ -51,16 +53,46 @@ public class RedisHubLifetimeManager<THub> : HubLifetimeManager<THub>, IDisposab
             _ackHandler = new AckHandler();
 
             var writer = new LoggerTextWriter(logger);
-            _logger.ConnectingToEndpoints(string.Join(", ", options.Value.Options.EndPoints.Select(e => EndPointCollection.ToString(e))));
+            _logger.ConnectingToEndpoints(options.Value.Options.EndPoints);
             _redisServerConnection = _options.Connect(writer);
+
+            _redisServerConnection.ConnectionRestored += (_, e) =>
```

Comment:
Shouldn't these events be wired before you call Connect? What if the ConnectionRestore event happens immediately after calling Connect and before the event is wired up?

Reply:
This is the earliest we get the connection

Reply:
Ah, yes. Now I remember. I never liked it.